pkg/apis/wsecurity/v1alpha1/body.go

```diff
@@ -0,0 +1,182 @@
+package v1alpha1
+
+import "fmt"
+
+//////////////////// BodyProfile ////////////////
+
+// Exposes ValueProfile interface
+type BodyProfile struct {
+	Unstructured *SimpleValProfile  `json:"unstructured"`
+	Structured   *StructuredProfile `json:"structured"`
+}
+
+func (profile *BodyProfile) profileI(args ...interface{}) {
+	switch v := args[0].(type) {
```

Comment:
Whenever I see casting in Golang I have to ask was it needed? Can you add some explation on why this is needed since the repercussion are severe leading to all types of issue.

So please add comments on why this is needed.

Reply:
See https://github.com/knative-sandbox/security-guard/pull/20
type assertions are used only for common testing and alignment of all data types. It is not exported outside of the package and is not used inside of the package except when testing.

Reply:
Hmm, I would then expect the switch statement to be in a test code...

Reply:
We have about 18 DataTypes right now and this number will grow. 
Alignment between these dataTypes is key to being able to maintain the code and grow it. 
Each data Type conforms to a common interface that is being used for the testing as the backbone for aligning all dataTypes. 

This is explained in #20 
Conforming to the interface means that the objects need to implement certain methods that can't move to the testing files. Also, I don't think this is the correct design for a class to have methods in test files. 

I understand that this may seem unorthodox, but the compromise here between maintaining a common interface for alignment and avoiding runtime type assertions, resulted in this design which we have now implemented to all 18 dataTypes. 

We are discussing the dataType structure for multiple weeks now, and this design is where we got so far. 
If we decide to change this design and make the change to all 18 dataTypes at any point, for now, this is the design we chose and until we agree on a new one let's move on and review all other aspects of the dataTypes here - see if we can merge the code. 

If we want to change the design, in view of the complexity of such a decision, I suggest that we discuss this f2f if needed. This is all documented in previous PRs already.

In any case, I prefer that we do not stop the progress with this repository now and if needed make the change later.